src/components/legend/handle_click.js

```diff
@@ -38,44 +38,75 @@ module.exports = function handleClick(g, gd, numClicks) {
     var legendItem = g.data()[0][0];
     if(legendItem.groupTitle && legendItem.noClick) return;
 
+    var i, j;
     var fullData = gd._fullData;
+    var allLegendItems = fullData.slice();
+    if(fullLayout.shapes) {
+        for(i = 0; i < fullLayout.shapes.length; i++) {
+            var shapeLegend = fullLayout.shapes[i]; // TODO: make a copy instead!
```

Comment:
Is this TODO still relevant? I guess so, as I don't think we want to be adding a non-underscore attribute (`index`) to the actual `fullLayout.shapes` entries.

Reply:
oh haha I was looking at an old version of the code, you already did this!